python_modules/dagster/dagster_tests/core_tests/instance_tests/test_instance.py

```diff
@@ -467,6 +469,44 @@ def test_create_run_with_partitioned_asset_stores_partitions_snapshot():
             )
         )
 
+        # single partition
+        run = create_run_for_test(
+            instance=instance,
+            job_name="foo",
+            execution_plan_snapshot=ep_snapshot,
+            job_snapshot=noop_time_window_asset_job.get_job_snapshot(),
+            tags={
+                PARTITION_NAME_TAG: "2025-1-1",
+            },
+            asset_graph=noop_time_window_asset_job.asset_layer.asset_graph,
+        )
+        partitions_def = noop_time_window_asset_job.asset_layer.asset_graph.get(
+            dg.AssetKey("noop_time_window_asset")
+        ).partitions_def
+        assert partitions_def is not None
+
+        assert run.partitions_subset is not None
+        assert run.partitions_subset == partitions_def.subset_with_partition_keys(
+            partitions_def.get_partition_keys_in_range(
+                PartitionKeyRange(
+                    "2025-1-1",
+                    "2025-1-1",
+                )
+            )
+        )
+
+        # a run created with no partition key but targeting a partitioned asset should not store a
+        # partitions subset
+        run = create_run_for_test(
```

Comment:
hmm this passes against master, figuring out why

Reply:
oh, the refactor of the instance module introduced a check on the start and end tag